spec/models/spree/stock/availability_validator_spec.rb

```diff
@@ -0,0 +1,75 @@
+require 'spec_helper'
+
+module Spree
+  module Stock
+    describe AvailabilityValidator do
+      let(:validator) { AvailabilityValidator.new({}) }
+
+      context "line item without existing inventory units" do
+        let(:order) { create(:order_with_line_items) }
+        let(:line_item) { order.line_items.first }
+
+        before do
+          expect(order.shipment.inventory_units).to be_empty
+          expect(line_item.target_shipment).to be_nil
+        end
+
+        context "available quantity when variant.on_hand > line_item.quantity" do
+          it "suceeds" do
+            line_item.quantity = line_item.variant.on_hand - 1
+            validator.validate(line_item)
+            expect(line_item.errors[:quantity].size).to eq(0)
+          end
+        end
+
+        context "unavailable quantity when variant.on_hand < line_item.quantity" do
+          it "fails" do
+            line_item.quantity = line_item.variant.on_hand + 1
+            validator.validate(line_item)
+            expect_product_out_of_stock_error
```

Comment:
what about replacing this with `expect(line_item).not_to be_valid`?

Reply:
done